packages/mui-joy/src/Snackbar/Snackbar.tsx

```diff
@@ -0,0 +1,376 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { ClickAwayListener } from '@mui/base/ClickAwayListener';
+import { useSnackbar } from '@mui/base/useSnackbar';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import { OverridableComponent } from '@mui/types';
+import { ColorInversionProvider } from '../styles/ColorInversion';
+import useSlot from '../utils/useSlot';
+import styled from '../styles/styled';
+import { useThemeProps } from '../styles';
+import { resolveSxValue } from '../styles/styleUtils';
+import { SnackbarProps, SnackbarOwnerState, SnackbarTypeMap } from './SnackbarProps';
+import { getSnackbarUtilityClass } from './snackbarClasses';
+
+const useUtilityClasses = (ownerState: SnackbarOwnerState) => {
+  const { variant, color, size, anchorOrigin } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      size && `size${capitalize(size)}`,
+      color && `color${capitalize(color)}`,
+      variant && `variant${capitalize(variant)}`,
+      `anchorOrigin${capitalize(anchorOrigin!.vertical)}${capitalize(anchorOrigin!.horizontal)}`,
+    ],
+    startDecorator: ['startDecorator'],
+    endDecorator: ['endDecorator'],
+  };
+
+  return composeClasses(slots, getSnackbarUtilityClass, {});
+};
+
+const SnackbarRoot = styled('div', {
```

Comment:
I think it could reuse the Alert slots.

```js
// Alert.tsx

export const StyledAlertRoot = styled('div')({ … });

// Snackbar.tsx
import { StyledAlertRoot } from '../Alert/Alert.tsx';

const SnackbarRoot = styled(StyledAlertRoot as unknown as 'div')({ … });
```

For more detail, you can take a look at IconButton and ModalClose components.

Reply:
I had the same idea. But if we use the name `--Alert` for Snackbar's underlying CSS variables, wouldn't it be wiser to create separate variables for Snackbar, even though their styles are the same as Alert's? I'm uncertain, but it might be useful when overriding styles in the global theme.